### DISCUSSION TITLE
Error when trying to populate entity later

### DISCUSSION THREAD

**Discussion by nik-lampe** [mikro-orm/mikro-orm#3349] [Q&A]
**Describe the bug**
I'm using mikro-orm together with TypeGraphQL to build a GraphQL API.
I load some entities in a Resolver and try to populate them in a field resolver at a later point.

**Stack trace**
```
TypeError: Cannot read properties of undefined (reading '__serializationContext')
    at /Users/nik/Code/BugReports/mikroorm/node_modules/@mikro-orm/core/entity/EntityLoader.js:35:42
    at Array.forEach (<anonymous>)
        at EntityLoader.populate (/Users/nik/Code/BugReports/mikroorm/node_modules/@mikro-orm/core/entity/EntityLoader.js:35:18)
        at SqlEntityManager.populate (/Users/nik/Code/BugReports/mikroorm/node_modules/@mikro-orm/core/EntityManager.js:680:31)
        at FooResolver.bar (/Users/nik/Code/BugReports/mikroorm/src/resolvers/FooResolver.ts:40:14)
        at /Users/nik/Code/BugReports/mikroorm/node_modules/type-graphql/dist/resolvers/create.js:34:68
        at Object.applyMiddlewares (/Users/nik/Code/BugReports/mikroorm/node_modules/type-graphql/dist/resolvers/helpers.js:58:16)
        at /Users/nik/Code/BugReports/mikroorm/node_modules/type-graphql/dist/resolvers/create.js:27:26
        at resolveField (/Users/nik/Code/BugReports/mikroorm/node_modules/graphql/execution/execute.js:464:18)
        at executeFields (/Users/nik/Code/BugReports/mikroorm/node_modules/graphql/execution/execute.js:292:18)"
```

**To Reproduce**
1. Clone this repo: https://github.com/nik-lampe/mikro-orm-bug
2. Run npm install
3. Run npm run dev
4. Go To localhost:4000/graphql
5. Execute this query:
```
{
  foos {
    id
    title
    bar {
      id
      title
    }
  }
}
```

**Expected behavior**
Well, it should populate the entity and then return it :) 

**Versions**

| Dependency | Version |
| - | - |
| node | 18.1 |
| typescript | 4.7.4 |
| mikro-orm | 5.2.4 |
| your-driver | @mikro-orm/sqlite and @mikro-orm/mariadb |

**Comment by nik-lampe** [comment]
Literally seconds later I found out about the WrappedEntity 😬 

I guess this is how it works? Sorry for this

**Reply by B4nan** [reply]
Looks like you are trying to populate your `Foo` class which is not an entity, that's all. 

edit: Will try to add some validation, this is indeed weird

**Reply by B4nan** [reply]
Actually, you have two classes called `Foo`, one is [an entity](https://github.com/nik-lampe/mikro-orm-bug/blob/main/src/entities/Foo.ts) and [one is not](https://github.com/nik-lampe/mikro-orm-bug/blob/main/src/resolvers/FooResolver.ts#L23-L32).

**Reply by nik-lampe** [reply]
I see. That makes sense, the object passed to the bar method is already an instance the Foo ObjectType.

Although I made the mistake and using em.find() on the ObjectType class declared in the same file, the same behaviour happens when using the entity class.

**Reply by nik-lampe** [reply]
Thank you 👍

**Reply by nik-lampe** [reply]
I guess that's why wrapping it won't help either?

**Reply by B4nan** [reply]
This won't happen if you use an entity class, as entity class will have that `__helper` property defined - they are added to the class prototype so even if you create that class via constructor, it will be there. I can't help much with GQL issues, but if you create a snippet without it, will be happy to comment on that :]

**Reply by B4nan** [reply]
FYI validation added [here](https://github.com/mikro-orm/mikro-orm/commit/ab93106fd104be3ae12efb7c2fa680ad6f348d27)
